### PR TITLE
Prevent logging exceptions twice when debugger is opened

### PR DESCRIPTION
**Pull request by hernanmd**
Change logDebuggerStackToFile default setting to false since when set to true, it makes the debug session log twice on every request as reported in #15620.

### CODE REVIEW

**Comment by guillep** [comment]
Maybe you would like to discuss this with @StevenCostiou ?

If Pharo is already logging exceptions, why should the debugger do it also?
My point here is that maybe the code of this setting should just be removed, but I'm not sure :).
Otherwise, changing the setting looks like it's not solving the cause of the problem but its consequence, right? People with a bad setting could also have a bad experience, and not understand where it comes from.
In other words: if the setting has no value, why having it?

**Comment by StevenCostiou** [comment]
After discussion, we propose to merge this PR and open an issue to specifically discuss the need for a second mechanism.
It is not clear right now if we do or not.
However not logging upon debugger opening can improve a bit the speed? - and we need to gain avery possible bit of speed in the debugger ;)

**Comment by hernanmd** [comment]
Ok, I created another issue for this: #15810

**Comment by Ducasse** [comment]
I merge as requeted.

**Comment by daniels220** [comment]
Dang, I've been thinking about saying something about this for a long time, just now getting to it—wish I'd slipped my thoughts in sooner, because:

I think, instead, the system should skip the _unconditional_ logging to the file, in `UIManager>>requestDebuggerOpeningFor:`, allowing the existing setting to do exactly what it says—control whether the stack is logged _at all_ when opening a debugger. In which case I think `true` is a sensible default. As a fallback we should also log in `requestDebuggerOpeningFor:` if `OupsDebuggerSystem` is not present.

**Comment by guillep** [comment]
> Dang, I've been thinking about saying something about this for a long time, just now getting to it—wish I'd slipped my thoughts in sooner, because:
> 
> I think, instead, the system should skip the _unconditional_ logging to the file, in `UIManager>>requestDebuggerOpeningFor:`, allowing the existing setting to do exactly what it says—control whether the stack is logged _at all_ when opening a debugger. In which case I think `true` is a sensible default. As a fallback we should also log in `requestDebuggerOpeningFor:` if `OupsDebuggerSystem` is not present.

I like the idea :)

**Comment by daniels220** [comment]
Actually, I'm going to kinda take that back—but I'm also going to put my actual comment on #15810, since that's the open issue.
